languages/rust/oso/src/host/class.rs

```diff
@@ -56,34 +73,38 @@ impl Default for Class {
     }
 }
 
-impl<T> Class<T> {
+// TODO seems like the name is based on fully qualified name, so we may want to
+// require this to be specified.
+
+impl<T> Class<T>
+where T: 'static
+{
     pub fn new() -> Self
-    where
-        T: 'static,
     {
+        let name = std::any::type_name::<T>().to_string();
         Self {
-            name: std::any::type_name::<Self>().to_string(),
+            name: name.clone(),
             constructor: None,
             attributes: InstanceMethods::new(),
             instance_methods: InstanceMethods::new(),
             class_methods: ClassMethods::new(),
             instance_check: Arc::new(|any| any.is::<T>()),
             class_check: Arc::new(|type_id| TypeId::of::<T>() == type_id),
+            equality_check: Arc::from(equality_not_supported(name)),
             ty: std::marker::PhantomData,
             type_id: TypeId::of::<T>(),
         }
     }
 
     pub fn with_default() -> Self
     where
-        T: std::default::Default + 'static,
+        T: std::default::Default,
```

Comment:
@samscott89 Why was static part of all of these before if we don't need it?

Reply:
It's needed, but moved up to the top:

```
impl<T> Class<T>
where T: 'static
{
```